lib/mof/scanner.rb

```diff
@@ -144,10 +144,10 @@ def fill_queue
 	when "sint64" then @q.push [:DT_SINT64, CIM::Type.new(m)]
 	when "sint8" then @q.push [:DT_SINT8, CIM::Type.new(m)]
```

Comment:
And on that note, I think it probably then makes sense to use some named captures to separate namespace from typename like:

```
[25] pry(main)> r = %r{(?<namespace>[\w\.]+\.)?(?<type>\w+)}
=> /(?<namespace>[\w\.]+\.)?(?<type>\w+)/
[26] pry(main)> 'system.foo'.match(r)
=> #<MatchData "system.foo" namespace:"system." type:"foo">
```

Note that that regex for namespace isn't totally perfect, since it would allow a match like `..foo`, but I think it's an OK tradeoff without going crazy.